New recipe: concat files of different types (+ a few fixups)

kfrn opened the pull request:
I was thinking about #292 today, and decided to look into the `concat` filter some more. Here I've added a recipe for concatenating input files of different types.

Questions/issues:
- Wording may need some love ... feel free to nitpick
- I'm getting the desired results, but if someone else can test the command(s), that would be super
- ~If you don't specify the specs for the output streams, as described in the second part of the entry, it's pretty unclear to me how ffmpeg 'decides' what the output specs will be - the [docs](https://ffmpeg.org/ffmpeg-filters.html#concat) aren't very clear on this point. (Actually, I find the concat docs confusingly worded in general).~ Lol I'm an idiot, see below. Though I still think the concat docs could be better worded just generally.
  - I've asked about this on the ffmpeg-user listserv ([link](http://ffmpeg.org/pipermail/ffmpeg-user/2018-January/038698.html)), hopefully someone there will have some more info.
  - Update: Ugh, this is what happens when you look at this stuff as insomniac. The default output specs are just those of the output file type - DUH. I don't know why I thought some extra \~magic\~ was going on there, but I got that idea in my head. Now my embarrassing misunderstanding complete with multiple typos is immortalised on the ffmpeg-user, great job Katherine.
- This recipe, though already lengthy, could be expanded. For example, it would be good to include a command for inputs of different resolution, e.g. using a scale filter prior to concatenation.

Also, a couple of little fixes elsewhere:
- add `-c:a aac` to the H.264/MP4 recipes. Since AAC and MP4 are birds of a feather, I think that's more sensible than having `-c:a copy`.
- add a missing line to a recipe explanation

kfrn committed:
Specify AAC as audio codec in H.264/MP4 recipes

kfrn committed:
Add missing line to recipe explanation

kfrn committed:
Add basic recipe to concat files of different types

kfrn committed:
Update concat diff files recipe to include information on transcoding

kfrn committed:
Amend wording; remove accidentally committed comment

retokromer commented on index.html:
```diff
@@ -905,7 +907,59 @@ <h3>Join files together</h3>
+      <p>
+        This command takes two or more files of the different file types and joins them together to make a single file.<br>
+        However, it only works properly if the files to be combined have the same dimensions (e.g., 720x576). Some aspects of the input files will be normalised: for example, if an input file contains a video track and an audio track that do not have exactly the same duration, the shorter one will be padded. However, note that if the input files have different framerates, the output file will be  ...
+      </p>
```
I would not change our coding style. Don’t add a newline after the `<p>` opening tag nor before the `</p>` closing tag.

retokromer commented on index.html:
```diff
@@ -905,7 +907,59 @@ <h3>Join files together</h3>
+      <p>
+        Likewise, to encode the output audio stream as mp3, the command could include the following:<br>
+        <code>-map "[audioOut]" -c:a libmp3lame -dither_method modified_e_weighted -qscale:a 2</code><br>
+      </p>
```
A `<br>` immediately before the closing tag `</p>` is useless.

kfrn replied:
Opps, the break tag was from when I had another line after that. Fixed in 73cd80b. Cheers @retokromer !

retokromer reviewed (commented):
My two nits.

kfrn committed:
Tidy HTML

kfrn committed:
Add more info about default behaviour for concat filter command

Also change the formatting a bit: separate paragraphs, as opposed to using break tags.

retokromer commented on index.html:
```diff
@@ -905,7 +907,53 @@ <h3>Join files together</h3>
+    </div>
+    <!-- ends Join files of the different types together -->
+
 
```
Only one empty line?

retokromer commented on index.html:
```diff
@@ -332,7 +333,7 @@ <h3>Create FFV1 Version 3 video in a Matroska container with framemd5 of input</
     <div class="hiding">
       <h3>Convert DVD to H.264</h3>
-      <p><code>ffmpeg -i concat:<i>input_file1</i>\|<i>input_file2</i>\|<i>input_file3</i> -c:v libx264 -c:a copy <i>output_file</i>.mp4</code></p>
+      <p><code>ffmpeg -i concat:<i>input_file1</i>\|<i>input_file2</i>\|<i>input_file3</i> -c:v libx264 -c:a aac <i>output_file</i>.mp4</code></p>
```
I suggest to not mix up naming styles: `input_file_2` etc.

retokromer commented on index.html:
```diff
@@ -905,7 +907,53 @@ <h3>Join files together</h3>
+        <dt>a=1</dt><dd>sets the number of output audio streams.<br>
+          Note that this must be equal to the number of audio streams selected from each segment.</dd>
+        <dt>[videoOut]</dt><dd>name of the concatenated output video stream. This is a variable name which you define, so you could call it something different, like “vOut”, “outv”, or “banana”.</dd>
+        <dt>[audioOut]</dt><dd>name of the concatenated output audio stream. Again, this is a variable name which you define.</dd>
```
I suggest to use `video_out`,  `audio_out` etc.

retokromer reviewed (commented):
A few styling nits.

retokromer reviewed (approved):
LGTM, I can open a follow PR to uniform naming through all the formulae.

privatezero commented:
I totally am going to be naming all my variables for mapping `[banana]` from now on.

kfrn commented:
Lol, for some reason I always use fruits as placeholder variable names. Mostly in the repl, but I'm sure there is some code somewhere where I left in an `apple` ...

GregH18 commented:
I know this is late in the game (novice GitHub user here), but thank you @kfrn for thinking about the concat/transcode functionality. I imagine @bturkus is already aware of this, but tagging him here just in case. Thanks again!

kieranjol commented:
+1
